src/TaskLineRenderer.ts

```diff
@@ -217,11 +240,14 @@ async function taskToHtml(
                 addInternalClasses(component, internalSpan);
 
                 // Add the component's CSS class describing what this component is (priority, due date etc.)
-                const componentClass = getTaskComponentClass(component);
-                span.classList.add(...componentClass);
+                const fieldLayoutDetails = FieldLayouts[component];
+                if (fieldLayoutDetails) {
+                    const componentClass = [fieldLayoutDetails.className];
+                    span.classList.add(...componentClass);
+                }
```

Comment:
FYI this check is related to our previous comments: https://github.com/obsidian-tasks-group/obsidian-tasks/pull/2403#discussion_r1389128258

Let me know if this can be removed now. TBH I don't see the danger here.

Reply:
then `fieldLayoutDetails` will be `null` and the code inside the `if` won't be executed.

Reply:
Let's talk about this next time we pair... We aren't going to be able to resolve by typing and going round in circles.

Reply:
> Let's talk about this next time we pair... We aren't going to be able to resolve by typing and going round in circles.

Resolving - I've added it to my notes for our next pairing session